test/groovy/DockerExecuteOnKubernetesTest.groovy

```diff
@@ -819,6 +819,48 @@ class DockerExecuteOnKubernetesTest extends BasePiperTest {
                 "name"     : "volume",
                 "mountPath": "/opt"
             ]], containerSpec.volumeMounts)
+        assertEquals(
+            [[
+                 "name"     : "volume",
+                 "mountPath": "/opt"
+             ]], initContainerSpec.volumeMounts)
+    }
+
+    @Test
+    void testInitContainerDefaultWithParameters() {
+        stepRule.step.dockerExecuteOnKubernetes(
+            script: nullScript,
+            juStabUtils: utils,
+            containerName: 'mycontainer',
+            initContainerImage: 'ppiper/cf-cli@sha256:latest',
+            initContainerCommand: 'cp /usr/local/bin/cf7 /opt/bin/cf',
+            dockerImage: 'maven:3.5-jdk-8-alpine',
+            containerMountPath: '/opt',
+        ) { bodyExecuted = true }
+        def initContainer = podSpec.spec.initContainers[0]
+        def expectedCommandOutput = ["sh", "-c", "cp /usr/local/bin/cf7 /opt/bin/cf"]
+        assertTrue(bodyExecuted)
+        assertEquals("ppiper-cf-cli-sha256-latest", initContainer.name)
+        assertEquals("ppiper/cf-cli@sha256:latest", initContainer.image)
+        assertTrue(initContainer.command.size() == expectedCommandOutput.size())
+        assertTrue(initContainer.command.containsAll(expectedCommandOutput))
+    }
+
+    @Test
+    void testInitContainerWithoutContainerCommand() {
+        stepRule.step.dockerExecuteOnKubernetes(
+            script: nullScript,
+            juStabUtils: utils,
+            containerName: 'mycontainer',
+            initContainerImage: 'ppiper/cf-cli@sha256:latest',
+            dockerImage: 'maven:3.5-jdk-8-alpine',
+            containerMountPath: '/opt',
+        ) { bodyExecuted = true }
+        def initContainer = podSpec.spec.initContainers[0]
+        def expectedCommandOutput = ["/usr/bin/tail", "-f", "/dev/null"]
+        assertTrue(bodyExecuted)
+        assertTrue(initContainer.command.size() == expectedCommandOutput.size())
+        assertTrue(initContainer.command.containsAll(expectedCommandOutput))
```

Comment:
size check for array is removed as suggested.

Reply:
How does containsAll work? Will it fail if the order is different? Will it fail if initContainer.command has additional values?

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/List.html#containsAll(java.util.Collection)
It will not fail if the order is different + it also doesn't check for duplicates.
`List<String> list1 = Arrays.asList("a", "a", "b")
List<String> list2 = Arrays.asList("a", "b", "b") 
assertEquals(list1.size(), list2.size())
assertTrue(list1.containsAll(list2) && list2.containsAll(list1))
`

Reply:
But the order is important. Please check differently that the lists are the same.

Reply:
Please take a look at the new commit.